src/WebJobs.Extensions.DurableTask/AzureStorageAccountProvider.cs

```diff
@@ -0,0 +1,106 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal sealed class AzureStorageAccountProvider : IStorageAccountProvider
+    {
+        private readonly IConnectionInfoResolver connectionInfoResolver;
+        private readonly IStorageCredentialsFactory credentialFactory;
+
+        public AzureStorageAccountProvider(IConnectionInfoResolver connectionInfoResolver, IStorageCredentialsFactory credentialFactory = null)
+        {
+            this.connectionInfoResolver = connectionInfoResolver ?? throw new ArgumentNullException(nameof(credentialFactory));
+            this.credentialFactory = credentialFactory ?? NullCredentialsFactory.Instance;
+        }
+
+        public StorageAccountDetails GetStorageAccountDetails(string name)
+        {
+            AzureStorageAccountOptions account = this.ResolveAccountInfo(name);
+            if (!string.IsNullOrEmpty(account.ConnectionString))
+            {
+                return new StorageAccountDetails { ConnectionString = account.ConnectionString };
+            }
+
+            StorageCredentials storageCredentials = this.credentialFactory.CreateAsync(account).EnsureCompleted();
+            if (storageCredentials == null)
+            {
+                throw new InvalidOperationException($"Unable to resolve the Azure Storage credentials for connection name '{name}'.");
+            }
+
+            if (account.QueueServiceUri != null || account.TableServiceUri != null)
+            {
+                // TODO: Use new endpoints when Durable Task is updated
+                return new StorageAccountDetails
+                {
+                    StorageCredentials = storageCredentials,
+                };
+            }
+            else
+            {
+                return new StorageAccountDetails
+                {
+                    AccountName = account.AccountName,
+                    EndpointSuffix = AzureStorageAccountOptions.DefaultEndpointSuffix,
+                    StorageCredentials = storageCredentials,
+                };
+            }
+        }
+
+        public CloudStorageAccount GetCloudStorageAccount(string name)
+        {
+            AzureStorageAccountOptions account = this.ResolveAccountInfo(name);
+            if (!string.IsNullOrEmpty(account.ConnectionString))
+            {
+                return CloudStorageAccount.Parse(account.ConnectionString);
+            }
+
+            StorageCredentials storageCredentials = this.credentialFactory.CreateAsync(account).EnsureCompleted();
+            if (storageCredentials == null)
+            {
+                throw new InvalidOperationException($"Unable to resolve the Azure Storage credentials for connection name '{name}'.");
+            }
+
+            if (account.QueueServiceUri != null || account.TableServiceUri != null)
```

Comment:
> @cgillum when is Blob storage used for Durable?

Durable uses blobs for lease operations and for storing payloads that don't fit in Azure Queues or Azure Tables.

> Are we ever only using a single storage account for all durable operations or do we use multiple (customer specified + AzureWebJobsStorage)

We only ever use a single storage account for all operations. We don't have any direct dependency on `AzureWebJobsStorage` (unlike some other triggers).

Reply:
Yes! I missed this. There was a brief period I had left blob storage out of the methods. But it's back now!
@karshinlin - I think this backfilling is the current behavior for WebJobs if I'm not mistaken. But I think it's because each of the service types is evaluated independently. In the DurableTask PR, I have change the backfill to actually throw if only a subset of the URIs are specified.